docs/src/convert-to-onda.md

```diff
@@ -0,0 +1,264 @@
+For storing `Onda.Samples`, see [`Onda.store`](https://beacon-biosignals.github.io/Onda.jl/stable/#Onda.store), which supports serializing LPCM-encoded samples to [any "path-like" type](https://beacon-biosignals.github.io/Onda.jl/stable/#Support-For-Generic-Path-Like-Types) (i.e., anything that provides a method for `write`).
+For storing the plan, use `Legolas.write(file_path, plan, FilePlanV2SchemaVersion())` (see the documentation for [`Legolas.write`](https://beacon-biosignals.github.io/Legolas.jl/stable/#Legolas.write) and [`FilePlanV2`](@ref).
+
+## Batch conversion of many EDFs
+
+The workflow for bulk conversion of multiple EDFs is similar to the workflow for converting a single EDF.
+The major difference is that the "planning" steps can be conducted in bulk, while the "execution" steps (generally) need to be conducted one at a time, either serially or distributed across multiple workers.
+As discussed above, the planning stage requires only a few KB from the EDF file/signal headers, facilitating rapid plan-review-revise iteration of even fairly large collections of EDFs (10,000+).
+
+### Planning multiple EDFs
+
+The main factor to consider when planning conversion of a large batch of EDF files is that planning requires only the (small number) of header bytes, even for very large EDF files.
+Thus, the first step is to read the file headers into memory without reading the signal data itself (which for more than a few EDF files will not in general fit into memory ).
+
+#### Reading headers from local filesystem
+
+For EDF files stored on a normal filesystem, the `EDF.File` constructor will by default create a "header-only" `EDF.File`, so multiple files' headers can be read like
+
+```julia
+files = map(edf_paths) do path
+    open(EDF.File, path, "r")
+end
+```
+
+#### Reading headers from S3
+
+!!! note
+    This section may become obsolete in a future version of EDF.jl which uses the [conditional dependency](https://pkgdocs.julialang.org/v1/creating-packages/#Weak-dependencies) functionality available from Julia 1.9+ to provide tighter integration with AWSS3.jl.
+
+Unfortunately, `open(path::S3Path)` will fetch the entire contents of the object stored at `path`, so we need to be a bit clever to read _only_ header bytes from an S3 file, especially given that the number of bytes we need to read depends on the number of signals.
+The following is an example of one technique for reading EDF file and signal headers from S3:
+
+```julia
+function EDF.read_file_header(path::S3Path)
+    bytes = s3_get(path.bucket, path.key; byte_range=1:256)
+    buffer = IOBuffer(bytes)
+    return EDF.read_file_header(buffer)
+end
+
+function EDF.File(path::S3Path)
+    _, n_signals = EDF.read_file_header(path)
+    bytes = s3_get(path.bucket, path.key; byte_range=1:(256 * (n_signals + 1)))
+    return EDF.File(IOBuffer(bytes))
+end
+
+# use asyncmap because this is mostly bound by request roundtrip latency
+files = asyncmap(EDF.File, edf_paths)
```

Comment:
Do we want to be a bit more general here and not use piracy(?) to define these functions, so that someone can easier copy and paste and use this?

e.g. (off the top of my head)
```suggestion
function _read_file_header(path::S3Path)
    bytes = s3_get(path.bucket, path.key; byte_range=1:256)
    buffer = IOBuffer(bytes)
    return EDF.read_file_header(buffer)
end

function _load_edf_file_header(path::S3Path)
    _, n_signals = _read_file_header(path)
    bytes = s3_get(path.bucket, path.key; byte_range=1:(256 * (n_signals + 1)))
    return EDF.File(IOBuffer(bytes))
end

# use asyncmap because this is mostly bound by request roundtrip latency
files = asyncmap(_load_edf_file_header, edf_paths)
```

Reply:
yeah, I originally had something like this (in the internal code I'm borrowing from) but wanted to make this as close to what I'd put in the EDF.jl package extension as possible...